src/test/ui/suggestions/option-to-bool.let_chains.stderr

```diff
@@ -0,0 +1,20 @@
+error[E0308]: mismatched types
+  --> $DIR/option-to-bool.rs:6:16
+   |
+LL |     if true && x {}
+   |                ^ expected `bool`, found enum `Option`
+   |
+   = note: expected type `bool`
+              found enum `Option<i32>`
+help: use `if let` to match on an `Option` in an `if` condition
```

Comment:
`Option` in a boolean position doesn't give any hint that the user wants to match anything inside the option, rather otherwise.
I don't think this should be suggested in this context.